pshtt/pshtt.py

```diff
@@ -651,106 +728,41 @@ def https_check(endpoint):
-        logging.exception("{}: Known error in sslyze 1.X with EC public keys. See https://github.com/nabla-c0d3/sslyze/issues/215".format(endpoint.url))
-        return None
-    except Exception as err:
-        endpoint.unknown_error = True
-        logging.exception("{}: Unknown exception in cert plugin.".format(endpoint.url))
-        utils.debug("{}: {}".format(endpoint.url, err))
-        return
-
-    # Debugging
-    # for msg in cert_response:
-    #     print(msg)
-
-    # Default endpoint assessments to False until proven True.
-    endpoint.https_expired_cert = False
-    endpoint.https_self_signed_cert = False
-    endpoint.https_bad_chain = False
-    endpoint.https_bad_hostname = False
-
-    # STORE will be either "Mozilla" or "Custom"
-    # depending on what the user chose.
-
-    # A certificate can have multiple issues.
-    for msg in cert_response:
-
-        # Check for missing SAN.
-        if (
-            (("DNS Subject Alternative Names") in msg) and
-            (("[]") in msg)
-        ):
-            endpoint.https_bad_hostname = True
-
-        # Check for certificate expiration.
-        if (
-            (STORE in msg) and
-            (("FAILED") in msg) and
-            (("certificate has expired") in msg)
-        ):
-            endpoint.https_expired_cert = True
-
-        # Check to see if the cert is self-signed
-        if (
-            (STORE in msg) and
-            (("FAILED") in msg) and
-            (("self signed certificate") in msg)
-        ):
-            endpoint.https_self_signed_cert = True
-
-        # Check to see if there is a bad chain
-
-        # NOTE: If this is the only flag that's set, it's probably
-        # an incomplete chain
-        # If this isnt the only flag that is set, it's might be
-        # because there is another error. More debugging would
-        # need to be done at this point, but not through sslyze
-        # because sslyze doesn't have enough granularity
-
-        if (
-            (STORE in msg) and
-            (("FAILED") in msg) and
-            (
-                (("unable to get local issuer certificate") in msg) or
-                (("self signed certificate") in msg)
-            )
-        ):
-            endpoint.https_bad_chain = True
-
-        # Check for whether the hostname validates.
-        if (
-            (("Hostname Validation") in msg) and
-            (("FAILED") in msg) and
-            (("Certificate does NOT match") in msg)
-        ):
-            endpoint.https_bad_hostname = True
+        logging.exception("{}: Unknown exception examining certificate deployment.".format(endpoint.url))
+        utils.debug("{}: Unknown exception examining certificate deployment: {}".format(endpoint.url, err))
 
     try:
-        endpoint.https_cert_chain_len = len(cert_plugin_result.received_certificate_chain)
+        endpoint.https_cert_chain_len = len(cert_plugin_result.certificate_deployments[0].received_certificate_chain)
```

Comment:
@jsf9k Thoughts on how we should handle multiple certificate deployments? Running this against the [current federal domain list](https://github.com/GSA/data/blob/master/dotgov-domains/current-federal.csv) there were at least ~50 domains that have two deployments.

Reply:
@mcdonnnj I can loop over all the certificate_deployments and add the length of received_certificate_chain. I think that should address this?

Reply:
@SaptakS This is another one that I'd like consensus on our end. The problem I see with just adding lengths is that it does not give clear insight into what the value means anymore. Does a value of `4` mean a single chain with a length of four, or does it mean two chains of length two?

Reply:
I'll add a mention for @h-m-f-t to get more thoughts.

Reply:
For now, I'm not especially bothered by the value representation as it stands. Simply adding the length would likely provide enough detail should we need to inspect longer chains because the outliers will be few and will be presented by a higher number.